src/logger.rs

```diff
@@ -0,0 +1,87 @@
+cfg_if::cfg_if! {
```

Comment:
I also do not see anything stm32h7 specific in this file (which is the only file that actually contains library code and not examples as far as I can see? ) so I don't believe this should live in stm32h7xx-hal, after all we are building a Hardware Abstraction Layer and this code does not exactly abstract hardware away, it provides some minimal logic around code that is already general across Cortex-M chips.